### PR TITLE
feat: add book shop demo

### PR DESCRIPTION
**Pull request by bodhisatan**
#### What type of PR is this?

feat
#### What this PR does / why we need it (en: English/zh: Chinese):

en: add ecom demo
zh: 添加电商demo

### CODE REVIEW

**Review by li-jin-gou — CHANGES_REQUESTED** [review]
use hz and idl to init api and add introduce to readme.md

File: book-shop/app/item/infras/repository/init.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		&gorm.Config{
+			PrepareStmt:            true,
+			SkipDefaultTransaction: true,
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	m := DB.Migrator()
```
Comment (li-jin-gou) [inline book-shop/app/item/infras/repository/init.go:49]: use `docker` init sql instead of migrate

File: book-shop/go.mod [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+require (
+	github.com/apache/thrift v0.13.0
+	github.com/cloudwego/kitex v0.4.3
+	github.com/hertz-contrib/swagger v0.0.0-20220711030440-b6402d4709f0
+	github.com/swaggo/files v0.0.0-20220728132757-551d4a08d97a
+)
+
+require (
+	github.com/bwmarrin/snowflake v0.3.0
```
Comment (li-jin-gou) [inline book-shop/go.mod:13]: I think we could use more extensions to improve the system such as otel、gzip、pprof

File: book-shop/go.mod [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+go 1.18
+
+require (
+	github.com/apache/thrift v0.13.0
+	github.com/cloudwego/kitex v0.4.3
+	github.com/hertz-contrib/swagger v0.0.0-20220711030440-b6402d4709f0
+	github.com/swaggo/files v0.0.0-20220728132757-551d4a08d97a
+)
+
+require (
```
Comment (GuangmingLuo) [inline book-shop/go.mod:12]: merge to one

File: book-shop/idl/item.thrift [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+// limitations under the License.
+//
+
+include "base.thrift"
+namespace go cwg.bookshop.item
+
+enum Status {
+    Online, // 上架
+    Offline, // 下架
+    Delete, // 删除
```
Comment (GuangmingLuo) [inline book-shop/idl/item.thrift:22]: 显示逗号的格式不太对

Comment (GuangmingLuo) [reply book-shop/idl/item.thrift:22]: 其它IDL文件也有类似问题

**Comment by GuangmingLuo** [comment]
1. 注意相关文件需要添加 license header，生成代码添加到 [path ignore](https://github.com/cloudwego/biz-demo/blob/main/.licenserc.yaml)
2. 在根目录 readme 添加子项目说明以及 maintainer

File: book-shop/go.mod [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	github.com/bwmarrin/snowflake v0.3.0
+	github.com/bytedance/sonic v1.7.0
+	github.com/cloudwego/hertz v0.3.2
+	github.com/gomodule/redigo v1.8.9
+	github.com/hertz-contrib/jwt v1.0.1
+	github.com/jinzhu/copier v0.3.5
+	github.com/kitex-contrib/registry-etcd v0.0.0-20221223084757-0d49e7162359
```
Comment (GuangmingLuo) [inline book-shop/go.mod:19]: you may use https://github.com/kitex-contrib/registry-etcd/releases/tag/v0.1.0 using etcd/server v3.5.7

File: book-shop/go.mod [hunk 6]
```diff
@@ -0,0 +1,89 @@
+module github.com/cloudwego/biz-demo/book-shop
+
+go 1.18
+
+require (
+	github.com/apache/thrift v0.13.0
+	github.com/bwmarrin/snowflake v0.3.0
+	github.com/bytedance/sonic v1.7.0
+	github.com/cloudwego/hertz v0.3.2
+	github.com/cloudwego/kitex v0.4.3
```
Comment (GuangmingLuo) [inline book-shop/go.mod:10]: update to the latest version: Hertz 0.5.1, Kitex 0.4.4

**Review by li-jin-gou — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**